test/test_fx.py

```diff
@@ -1597,6 +1598,62 @@ def test_no_mutation(self):
         with self.assertRaisesRegex(NotImplementedError, "new_args"):
             x[0] = 4
 
+    def test_custom_traceback_raised_when_exception_source_is_graphmodule(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.W = torch.nn.Parameter(torch.randn(5))
+
+            def forward(self, x):
+                return torch.dot(self.W, x)
+
+        traced = torch.fx.symbolic_trace(M())
+
+        out_nodes = [n for n in traced.graph.nodes if n.op == "output"]
+        out = out_nodes[-1]
+        relu_out = traced.graph.call_method(method_name='relu', args=(out.args[0],))
+        out.replace_all_uses_with(relu_out)
+
+        traced.recompile()
+
+        with self.capture_stderr() as captured:
+            with self.assertRaises(TypeError):
+                traced(5)
+
+        self.assertIn("Call using an FX-traced Module, line 4 of the "
+                      "traced Module’s generated forward function:",
+                      captured[0])
+
+    def test_custom_traceback_not_raised_when_exception_source_is_submodule(self):
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(3, 4)
+
+            def forward(self, x):
+                return self.linear(x)
+
+        traced = torch.fx.symbolic_trace(M())
+
+        # TODO: Figure out why using `capture_stderr` as a context
+        # manager doesn't work in this case even though it works on the
+        # other `test_custom_traceback` test. I'd put the "woman
+        # shrugging" emoji in here but I don't want to press my luck
+        # in terms of things unexpectedly failing
```

Comment:
I think it's because when you intercept the exception propagation and print out the GraphModule context, that happens in a a callee frame with respect to this test, but when the exception isn't intercepted it propagates out through here, exits the test, and prints to `stderr` in some test harness function (caller of this function)